wagtail/blocks/stream_block.py

```diff
@@ -429,37 +421,58 @@ class RawDataView(MutableSequence):
         and will not be saved back when calling get_prep_value.
         """
 
-        def __init__(self, stream_value):
+        def __init__(self, stream_value, use_raw=False):
             self.stream_value = stream_value
+            if use_raw:
+                self.data = stream_value._raw_data
+                self.is_raw = True
+            else:
+                self.data = stream_value._data
+                self.is_raw = False
+            self._initial_data_hash = hash(repr(self.data))
```

Comment:
Does anyone know if this would be problematic for different value types? I've only ever really seen dicts, lists, strings, ints and floats in stream values. Am I being naive here?